scls-format/src/Cardano/SCLS/Internal/Reader.hs

```diff
@@ -52,12 +61,31 @@ withNamespacedData filePath namespace f =
 
 This function does not provide additional checks.
 -}
-extractRootHash :: FilePath -> IO (Maybe Digest)
-extractRootHash filePath = do
+extractRootHash :: FilePath -> IO Digest
+extractRootHash = withLatestManifestFrame \Manifest{..} ->
+  pure rootHash
+
+extractNamespaceHash :: Text -> FilePath -> IO (Maybe Digest)
+extractNamespaceHash ns = withLatestManifestFrame \Manifest{..} ->
+  pure (namespaceHash <$> Map.lookup ns nsInfo)
+
+data NotSCLSFile = NotSCLSFile
+  deriving (Show, Typeable, Exception)
+
+withLatestManifestFrame :: (Manifest -> IO r) -> FilePath -> IO r
+withLatestManifestFrame f filePath = do
   IO.withBinaryFile filePath ReadMode \handle -> do
     h <- hFileSize handle
-    hSeek handle AbsoluteSeek (h - fromIntegral hashDigestSize)
-    bs <- BS.hGet handle (fromIntegral hashDigestSize)
-    case digestFromByteString bs of
-      Just d -> pure (Just d)
-      Nothing -> fail "Invalid digest"
+    hSeek handle AbsoluteSeek (h - 4)
+    bs <- BS.hGet handle 4
+    offset <- case runGetOrFail getWord32be (BSL.fromStrict bs) of
+      Right (_, _, d) -> return d
+      Left{} -> throwIO NotSCLSFile
+    frameData <- fetchOffsetFrame handle (FrameView (5 + offset) 0x01 (fromIntegral h - fromIntegral offset - 5))
```

Comment:
Did we end up agreeing on adding some named values for this frame record types constants?